vpr/src/pack/re_cluster_util.h

```diff
@@ -32,52 +33,73 @@ ClusterBlockId atom_to_cluster(const AtomBlockId& atom);
-                              ClusterBlockId& old_clb,
-                              t_clustering_data& clustering_data,
-                              int& imacro,
-                              bool during_packing);
+bool remove_mol_from_cluster(const t_pack_molecule* molecule,
+                             int molecule_size,
+                             ClusterBlockId& old_clb,
+                             t_lb_router_data*& router_data);
 
 /**
  * @brief A function that starts a new cluster for one specific molecule
  * 
- * It place the molecule in a specific type and mode that should be passed by
+ * It places the molecule in a specific type and mode that should be passed by
  * the higher level routine.
  */
-bool start_new_cluster_for_atom(const AtomBlockId atom_id,
-                                const enum e_pad_loc_type& pad_loc_type,
-                                const t_logical_block_type_ptr& type,
-                                const int mode,
-                                const int feasible_block_array_size,
-                                int& imacro,
-                                bool enable_pin_feasibility_filter,
-                                ClusterBlockId clb_index,
-                                t_lb_router_data** router_data,
-                                std::vector<t_lb_type_rr_node>* lb_type_rr_graphs,
-                                PartitionRegion& temp_cluster_pr,
-                                t_clustering_data& clustering_data,
-                                bool during_packing);
+bool start_new_cluster_for_mol(t_pack_molecule* molecule,
+                               const t_logical_block_type_ptr& type,
+                               const int mode,
+                               const int feasible_block_array_size,
+                               bool enable_pin_feasibility_filter,
+                               ClusterBlockId clb_index,
+                               bool during_packing,
+                               t_clustering_data& clustering_data,
+                               t_lb_router_data** router_data,
+                               PartitionRegion& temp_cluster_pr);
 
 /**
  * @brief A function that fix the clustered netlist if the move is performed
  * after the packing is done and clustered netlist is built
+ *
+ * If you are changing clustering after packing is done, you need to update the clustered netlist by
+ * deleting the newly absorbed nets and creating nets for the atom nets that become unabsorbed. It also
+ * fixes the cluster ports for both the old and new clusters.
  */
-void fix_clustered_netlist(const AtomBlockId& atom_id,
+void fix_clustered_netlist(t_pack_molecule* molecule,
+                           int molecule_size,
                            const ClusterBlockId& old_clb,
                            const ClusterBlockId& new_clb);
 
+/**
+ * @brief A function that commits the molecule move if it is legal
```

Comment:
Explain the inputs (e.g. new_clb_created).  
Explain what commiting the molecule means (what is updated)?
I also suggest putting inputs first and then outputs that are updated via references; a bit easier to figure out input and output that way.